record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java

```diff
@@ -80,11 +79,16 @@ class InternalRecordBuilderProcessor {
         constructorVisibilityModifier = metaData.publicBuilderConstructors() ? Modifier.PUBLIC : Modifier.PRIVATE;
         initializers = recordFacade.initializers();
 
-        builder = TypeSpec.classBuilder(builderClassType.name()).addAnnotation(generatedRecordBuilderAnnotation)
-                .addModifiers(metaData.builderClassModifiers()).addTypeVariables(typeVariables);
+        builder = TypeSpec.classBuilder(builderClassType.name()).addModifiers(metaData.builderClassModifiers())
+                .addTypeVariables(typeVariables);
         if (metaData.addClassRetainedGenerated()) {
             builder.addAnnotation(recordBuilderGeneratedAnnotation);
         }
+        switch (metaData.addGeneratedAnnotation()) {
+        case ALWAYS -> builder.addAnnotation(generatedRecordBuilderAnnotation);
+        case NEVER -> {
+            /* no-op */ }
+        }
```

Comment:
Let's move this to a utility as it's always the same. The utility might have ctors/factories for different builder types.